### ISSUE TITLE
Grafana Agent OTLP Logs 2 Loki don't capture `InstrumentationScope`

### ISSUE THREAD

**Issue by cyrille-leclerc** [grafana/agent#3455]
# Environment

* Grafana Agent v0.32.1 on MAcOS
* Grafana Cloud
* Logs produced by otel-java auto instr 1.24.0 / logback-instrumentation then channel through OTel Collector to capture the log message in stdout thanks to the `logging` exporter

# Problem Description

OTLP logs that are exported to Loki lose the `InstrumentationScope` which is critical as it holds the `logger.name`.

### Log message (Loki through Grafana Explore)

`InstrumentationScope` is missing

```json
{
   "body":"SUCCESS placeOrder orderId=1599564 customerId=customer-12 price=3.0 paymentMethod=credit_cart shippingMethod=express shippingCountry=GB durationInNanos=178947750",
   "traceid":"7f878d343db739d7f54b8476c3ce8350",
   "spanid":"9389a7a72c5328c7",
   "severity":"INFO",
   "resources":{
      "deployment.environment":"production",
      "host.arch":"aarch64",
      "host.name":"host-1",
      "os.description":"Mac OS X 13.3",
      "os.type":"darwin",
      "process.command_line":"/opt/homebrew/Cellar/openjdk/19.0.2/libexec/openjdk.jdk/Contents/Home/bin/java -Xlog:gc*=info:file=/usr/local/var/log/my-shopping-cart/frontent_gc.log -javaagent:./../.otel/opentelemetry-javaagent-1.24.0.jar -Dserver.port=8080",
      "process.executable.path":"/opt/homebrew/Cellar/openjdk/19.0.2/libexec/openjdk.jdk/Contents/Home/bin/java",
      "process.pid":69293,
      "process.runtime.description":"Homebrew OpenJDK 64-Bit Server VM 19.0.2",
      "process.runtime.name":"OpenJDK Runtime Environment",
      "process.runtime.version":"19.0.2",
      "service.instance.id":"instance-1",
      "service.name":"frontend",
      "service.namespace":"com-shoppingcart",
      "service.version":"1.0-SNAPSHOT",
      "telemetry.auto.version":"1.24.0",
      "telemetry.sdk.language":"java",
      "telemetry.sdk.name":"opentelemetry",
      "telemetry.sdk.version":"1.24.0"
   }
}
```

### Log message (OTel Collector Logging Exporter)

```
2023-04-04T22:53:51.248+0200	info	ResourceLog #0
Resource SchemaURL: https://opentelemetry.io/schemas/1.18.0
Resource attributes:
     -> deployment.environment: Str(production)
     -> host.arch: Str(aarch64)
     -> host.name: Str(host-1)
     -> os.description: Str(Mac OS X 13.3)
     -> os.type: Str(darwin)
     -> process.command_line: Str(/opt/homebrew/Cellar/openjdk/19.0.2/libexec/openjdk.jdk/Contents/Home/bin/java -Xlog:gc*=info:file=/usr/local/var/log/my-shopping-cart/frontent_gc.log -javaagent:./../.otel/opentelemetry-javaagent-1.24.0.jar -Dserver.port=8080)
     -> process.executable.path: Str(/opt/homebrew/Cellar/openjdk/19.0.2/libexec/openjdk.jdk/Contents/Home/bin/java)
     -> process.pid: Int(69293)
     -> process.runtime.description: Str(Homebrew OpenJDK 64-Bit Server VM 19.0.2)
     -> process.runtime.name: Str(OpenJDK Runtime Environment)
     -> process.runtime.version: Str(19.0.2)
     -> service.instance.id: Str(instance-1)
     -> service.name: Str(frontend)
     -> service.namespace: Str(com-shoppingcart)
     -> service.version: Str(1.0-SNAPSHOT)
     -> telemetry.auto.version: Str(1.24.0)
     -> telemetry.sdk.language: Str(java)
     -> telemetry.sdk.name: Str(opentelemetry)
     -> telemetry.sdk.version: Str(1.24.0)
ScopeLogs #0
ScopeLogs SchemaURL:
InstrumentationScope com.mycompany.ecommerce.controller.OrderController
LogRecord #0
ObservedTimestamp: 1970-01-01 00:00:00 +0000 UTC
Timestamp: 2023-04-04 20:53:51.127 +0000 UTC
SeverityText: INFO
SeverityNumber: Info(9)
Body: Str(SUCCESS placeOrder orderId=1599564 customerId=customer-12 price=3.0 paymentMethod=credit_cart shippingMethod=express shippingCountry=GB durationInNanos=178947750)
Trace ID: 7f878d343db739d7f54b8476c3ce8350
Span ID: 9389a7a72c5328c7
Flags: 1
	{"kind": "exporter", "data_type": "logs", "name": "logging/debug"}

```

**Comment by mar4uk** [comment]
`InstrumentationScope` support was [added](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17071) to the loki exporter in [v0.68.0](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CHANGELOG.md#v0680) version of the collector-contrib. Agent uses v0.63.0 version, probably that's why `InstrumentationScope` is missing there

**Comment by tpaschalis** [comment]
cc @ptodev as one thing to keep in mind as we're upgrading our OTel dependency

**Comment by cyrille-leclerc** [comment]
Problem fixed with Grafana Agent Flow 0.35.0

![image](https://github.com/grafana/agent/assets/459691/6ca0d1a9-40fa-4433-99b0-478a7957526c)
